.github/workflows/unittest.yml

```diff
@@ -0,0 +1,91 @@
+name: "UnitTests"
+
+on:
+  # add push to main trigger for sonarcloud scan
+  push:
+    branches:
+      - "main"
+      - "master"
+
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - "nuvla/**"
+      - "tests/**"
+      - ".github/workflows/unittest.yml"
+
+
+
+jobs:
+  run-tests:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python_version: ["3.8", "3.9", "3.10", "3.11"]
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python_version }}
+          cache: 'pip'
+
+      - name: Load cached Poetry installation
+        id: cached-poetry
+        uses: actions/cache@v4
+        with:
+          path: ~/.local  # the path depends on the OS
+          key: poetry-${{ matrix.python_version }}  # increment to reset cache
+
+      - name: Setup Poetry
+        uses: snok/install-poetry@v1
+        if: steps.cached-poetry.outputs.cache-hit != 'true'
+        with:
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+
+      - name: Install dependencies
+        run: |
+          poetry install --with=tests
+
+      - name: Run Unittests
+        run: >
+          poetry run pytest 
+          --junitxml=test-report.xml 
+          --cov=nuvla 
+          --cov-report=xml 
+          --cov-report term 
+          --cov-config=.coveragerc 
+          --cov-branch
+
+      - name: Install jq (For Quality Gate)
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y jq
+
+
+# Temporary disabled until we solve the issue with the sonarcloud token
+      - name: SonarCloud Scan
+        if: matrix.python_version == '3.11'
```

Comment:
@ignacio-penas If possible I would define the following global env vars
```
PYTHON_VER_MATRIX=["3.9", "3.10", "3.11"]
PYTHON_VER_LAST=$(last ${PYTHON_VER_MATRIX}) # I'm obviously hand-waving here
```
and then using it in the matrix definition list 
```
python_version: ${PYTHON_VER_MATRIX}
``` 
and then use `${PYTHON_VER_LAST}` here and a bit below.

Reply:
@konstan github workflows does not allow global env vars (Only if defined from the repository it self) and when defining envs within a job, the allowed types are string, float and int. So I think this would overcomplicate things.

So I'd say we either do it with repository (or organization) level variables. Or leave it as is.

For the first approach, this is the implementation done to build the matrix on the validation workflows:
https://github.com/nuvlaedge/nuvlaedge/blob/c24f77632cfc1f7480719ca908e92ed61fcf9f2c/.github/workflows/validation.yml#L43-L63

Reply:
This was my suspicion when I was commenting. What I don't like is `"3.11"` is all over the "place". So, I tried to suggest a refactor. But if, as you say, the vector definitions are not possible, I would still opt for defining the `last` (highest supported) version number as I see it's used in a number of places.

Reply:
Solved using the repository level variable: SUPPORTED_PYTHON_VERSIONS. Then, using jq to extract the last value of the array.
So now updating supported python version versions would require updating the above mentioned variable and pyproject.toml file.